tests/Endpoints/TenantTokenTest.php

```diff
@@ -0,0 +1,171 @@
+        $this->assertArrayHasKey('offset', $response->toArray());
+        $this->assertArrayHasKey('limit', $response->toArray());
+        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+        $this->assertSame(7, $response->getNbHits());
+        $this->assertCount(7, $response->getHits());
+    }
+
+    public function testGenerateTenantTokenWithFilter(): void
+    {
+        $promise = $this->client->index('tenantToken')->addDocuments(self::DOCUMENTS);
+        $this->client->waitForTask($promise['uid']);
+        $promiseFromFilter = $this->client->index('tenantToken')->updateFilterableAttributes([
+            'id',
+        ]);
+        $this->client->waitForTask($promiseFromFilter['uid']);
+
+        $token = $this->privateClient->generateTenantToken((object) ['tenantToken' => (object) ['filter' => 'id > 10']]);
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('offset', $response->toArray());
+        $this->assertArrayHasKey('limit', $response->toArray());
+        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+        $this->assertSame(4, $response->getNbHits());
+        $this->assertCount(4, $response->getHits());
+    }
+
+    public function testGenerateTenantTokenWithSearchRulesOnOneIndex(): void
+    {
+        $this->createEmptyIndex('tenantTokenDuplicate');
+        $token = $this->privateClient->generateTenantToken(['tenantToken']);
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+        $this->expectException(ApiException::class);
+        $response = $tokenClient->index('tenantTokenDuplicate')->search('');
+    }
+
+    public function testGenerateTenantTokenWithApiKey(): void
+    {
+        $options = [
+            'apiKey' => $this->privateKey,
+        ];
+        $token = $this->client->generateTenantToken(['*'], $options);
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('offset', $response->toArray());
+        $this->assertArrayHasKey('limit', $response->toArray());
+        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+    }
+
+    public function testGenerateTenantTokenWithExpiresAt(): void
+    {
+        /* @phpstan-ignore-next-line */
```

Comment:
👀

Reply:
It didn't happen on local but in the CI `phpstan` don't like `DateTime` class and I get this error:
```
  7      Class DateTime referenced with incorrect case: Datetime.
```
So I add a comment to ignore the three lines with `DateTime` and I find this [issue](https://github.com/phpstan/phpstan/issues/3815)